### PR TITLE
Added SVG graphics support

### PR DESCRIPTION
**Pull request by naueramant**

### CODE REVIEW

**Comment by naueramant** [comment]
ping

**Comment by naueramant** [comment]
@abcum ?

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/naueramant/pdflib
```
Comment (tobiemh) [inline go.mod]:
```suggestion
module github.com/abcum/pdflib
```

File: include/golang.c [hunk 2]
```diff
@@ -349,6 +349,22 @@ void _PDF_initgraphics(PDF *p) {
...
+}
+
+void _PDF_fit_graphics(PDF *p, int graphics, double x, double y, const char *optlist) {
+	PDF_TRY(p) { PDF_fit_graphics(p, graphics, x, y, optlist); } PDF_CATCH(p) { }
+}
+
+double _PDF_info_graphics(PDF *p, int graphics, const char *keyword, const char *optlist) {
+	PDF_TRY(p) { return PDF_info_graphics(p, graphics, keyword, optlist); } PDF_CATCH(p) { }
+}
+
```
Comment (tobiemh) [inline include/golang.c:367]: Could these method definitions be in alphabetical order, among all the other method definitions?

File: include/golang.h [hunk 3]
```diff
@@ -178,6 +178,14 @@ double _PDF_info_textline(PDF *p, const char *text, int len, const char *keyword
...
 void _PDF_initgraphics(PDF *p);
 
+int _PDF_load_graphics(PDF *p, const char *type, const char *filename, int len, const char *optlist);
+
+void _PDF_close_graphics(PDF *p, int graphics);
+
+void _PDF_fit_graphics(PDF *p, int graphics, double x, double y, const char *optlist);
+
+double _PDF_info_graphics(PDF *p, int graphics, const char *keyword, const char *optlist);
+
```
Comment (tobiemh) [inline include/golang.h:188]: Could these method definitions be in alphabetical order, among all the other method definitions?

File: include/pdflib.h [hunk 4]
```diff
@@ -681,6 +681,22 @@ PDF_info_textline(PDF *p, const char *text, int len, const char *keyword,
...
+PDF_close_graphics(PDF *p, int graphics);
+
+/* Place vector graphics on a content stream, subject to various options. */
+PDFLIB_API void PDFLIB_CALL
+PDF_fit_graphics(PDF *p, int graphics, double x, double y, const char *optlist);
+
+/* Format vector graphics and query metrics and other properties. */
+PDFLIB_API double PDFLIB_CALL
+PDF_info_graphics(PDF *p, int graphics, const char *keyword, const char *optlist);
+
```
Comment (tobiemh) [inline include/pdflib.h:699]: Could these method definitions be in alphabetical order, among all the other method definitions?

File: include/pdflib.h [hunk 5]
```diff
@@ -1326,6 +1342,13 @@ struct PDFlib_api_s {
                 const char *keyword, const char *optlist);
     void (PDFLIB_CALL * PDF_initgraphics)(PDF *p);
     void (PDFLIB_CALL * PDF_lineto)(PDF *p, double x, double y);
+    int (PDFLIB_CALL * PDF_load_graphics)(PDF *p, const char *type, const char *filename,
+                int len, const char *optlist);
+    void (PDFLIB_CALL * PDF_close_graphics)(PDF *p, int graphics);
+    void (PDFLIB_CALL * PDF_fit_graphics)(PDF *p, int graphics, double x, double y,
+                const char *optlist);
+    double (PDFLIB_CALL * PDF_info_graphics)(PDF *p, int graphics, const char *keyword,
+                const char *optlist);
```
Comment (tobiemh) [inline include/pdflib.h:1351]: Could these method definitions be in alphabetical order, among all the other method definitions?

File: main.go [hunk 6]
```diff
@@ -511,6 +511,30 @@ func (p *PDFlib) InitGraphics() error {
...
+	C._PDF_fit_graphics(p.val, C.int(graphics), C.double(x), C.double(y), C.CString(options))
+	return p.catch()
+}
+
+// InfoGraphics ...
+func (p *PDFlib) InfoGraphics(graphics int, keyword string, options string) (float64, error) {
+	ret := float64(C._PDF_info_graphics(p.val, C.int(graphics), C.CString(keyword), C.CString(options)))
+	return ret, p.catch()
+}
+
```
Comment (tobiemh) [inline main.go:537]: Could these method definitions be in alphabetical order, among all the other method definitions?

**Review by tobiemh — CHANGES_REQUESTED** [review]
Hi @naueramant, apologies for missing this pull request!

Just a few minor changes, but otherwise I'll get this merged today.

**Comment by naueramant** [comment]
> Hi @naueramant, apologies for missing this pull request!
> 
> Just a few minor changes, but otherwise I'll get this merged today.

I will refactor the files asap 😄

**Pull request closed without merging**